Create ui-c8y-1021-64-0-as-a-solution-architect-i-want-to-see-more-details-of.md (#3255)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1021-64-0-as-a-solution-architect-i-want-to-see-more-details-of.md

---
date: ""
title: As a solution architect, I want to see more details of the plugin I want to install as readme for particular plugin.
product_area: Platform services
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62166
version: 1021.64.0
---

> As a solution architect, I want to see more details of the plugin I want to install as readme for particular plugin.

Reviewer: ```suggestion
Until now, readme file was possible to be added only once, to the whole package. All you had to do is to put README.md file in root folder of package. This new feature adds possibility to provide README.md file for particular plugins in package. 

To add readme for the plugin, README.md file needs to be provided in plugin codebase (recommended place is the same folder as exposed plugin module) and `readmePath` property (indicating path to this file) needs to be added to exports, for example:
```ts
{
  name: 'Example widget plugin',
  module: 'WidgetPluginModule',
  path: './src/app/widget/widget-plugin.module.ts',
  readmePath: './src/app/widget/README.md',
  description: 'Adds a custom widget to the shell application'
}
```
When package is deployed, readme can be viewed in package details view, package versions view (readme for specific version), installed plugins list and available plugins list to be installed.
```

---
date: ""

> title: As a solution architect, I want to see more details of the plugin I want to install as readme for particular plugin.

Reviewer 2: ```suggestion
title: Added option to provide a readme file for particular plugins in a package
```

product_area: Platform services
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62166
version: 1021.64.0
---

> Previously, the readme file could only be added once to the whole package by storing the *README.md* file in the root folder of the package. This feature adds the option to provide a readme file for particular plugins in the package. 

Reviewer 3: I woudl propose to add more context.  Perhaps:  Currently users have to access the readme for a package in order to understand what capabilities a particular plugin within that package provides.  With this update a readme file can be created for each plugin within a package, thus providing users with a better experience and placing the information closer to the plugin.

Reviewer 2: Slightly modified and aded to the file.

> Until now, readme file was possible to be added only once, to the whole package. All you had to do is to put README.md file in root folder of package. This new feature adds possibility to provide README.md file for particular plugins in package. 

Reviewer 2: ```suggestion
Previously, the readme file could only be added once to the whole package by storing the *README.md* file in the root folder of the package. This feature adds the option to provide a readme file for particular plugins in the package. 
```

> To add readme for the plugin, README.md file needs to be provided in plugin codebase (recommended place is the same folder as exposed plugin module) and `readmePath` property (indicating path to this file) needs to be added to exports, for example:

Reviewer 2: ```suggestion
To add the readme for the plugin, the **README.md** file must be provided in the plugin codebase (the recommended place is the same folder as the exposed plugin module) and  the`readmePath` property (indicating the path to this file) must be added to the exports. For example:
```

```ts
{
  name: 'Example widget plugin',
  module: 'WidgetPluginModule',
  path: './src/app/widget/widget-plugin.module.ts',
  readmePath: './src/app/widget/README.md',
  description: 'Adds a custom widget to the shell application'
}
```

> When package is deployed, readme can be viewed in package details view, package versions view (readme for specific version), installed plugins list and available plugins list to be installed.

Reviewer 2: ```suggestion
When package is deployed, readme can be viewed in package details view, package versions view (readme for specific version), installed plugins list and available plugins list to be installed.
```
```suggestion
When the package is deployed, the readme can be viewed in the package details, the package versions view (readme for specific version), the installed plugins list and the available plugins list.
```
